mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -278,6 +288,11 @@ class Word2Vec extends Serializable with Logging {
       new Iterator[Array[Int]] {
         def hasNext: Boolean = iter.hasNext
 
+        def sampling(freq: Double): Boolean = {
+          val p = math.max(0.0, 1.0 - math.sqrt(sample / freq))
+          Utils.random.nextDouble() > p
```

Comment:
This makes the result indeterministic. Could you fix a random seed? FYI, there already exists an issue with indeterministic behavior of Word2Vec caused by shuffle. Usually this leads to flaky tests. It would be nice to have deterministic result for reproducibility.